File: rust/lib/src/testsuite/testsuite.rs
use std::{collections::HashMap};

use super::{dyn_test::{DynTest}};

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
pub trait TestSuite {
	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	// NOTE: Due to Rust peculiarities, we need to return a result with instances of the `DynTest` object
	// rather than `Test`
	fn get_tests(&self) -> HashMap<String, Box<dyn DynTest>>;

	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	fn get_network_width_bits(&self) -> u32;
}

File: golang/lib/testsuite/test_suite.go
/*
 * Copyright (c) 2020 - present Kurtosis Technologies LLC.
 * All Rights Reserved.
 */

package testsuite

import "github.com/kurtosis-tech/kurtosis-client/golang/lib/services"

// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
type TestSuite interface {
	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	GetTests() map[string]Test

	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	GetNetworkWidthBits() uint32

	// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
	GetStaticFiles() map[services.StaticFileID]string
}
